client/my-sites/design-preview/index.js

```diff
@@ -165,36 +164,53 @@ const DesignPreview = React.createClass( {
 			return;
 		}
 		event.preventDefault();
-		// TODO: if the href is on the current site, load the href as a preview and fetch markup for that url
 	},
 
-	getPreviewUrl( site ) {
-		if ( ! site ) {
+	getParsedPreviewUrl( site ) {
+		if ( ! site && ! this.props.previewUrl ) {
 			return null;
 		}
+		const baseUrl = this.props.previewUrl ? this.props.previewUrl : site.options.unmapped_url;
+		const parsed = url.parse( baseUrl, true );
+		delete parsed.search;
+		return parsed;
+	},
 
-		const parsed = url.parse( site.options.unmapped_url, true );
+	getPreviewUrl( site ) {
+		const parsed = this.getParsedPreviewUrl( site );
+		if ( ! parsed ) {
+			return null;
+		}
 		parsed.query.iframe = true;
 		parsed.query.theme_preview = true;
-		if ( site.options && site.options.frame_nonce ) {
-			parsed.query['frame-nonce'] = site.options.frame_nonce;
+		if ( site && site.options.frame_nonce ) {
```

Comment:
I was curious about that too... but I thought that since the condition didn't exist higher up it seemed unlikely. I'll add a `get`. This is the kind of place a selector would come in handy.